python/foglamp/services/core/api/service.py

```diff
@@ -72,6 +72,8 @@ def get_service_records():
 
     try:
         data = await request.json()
+        if not isinstance(data, dict):
+            raise ValueError('Data payload must be a dictionary')
```

Comment:
This is not a very meaningful error to give a user. They are using a REST API and expressing the payload in JSON so the message should really be a message about JSON.